src/Ast.ml

```diff
@@ -88,18 +88,32 @@ let parens_kind i =
         Some (txt, opn, cls)
     | _ -> None
 
-let index_op_get i =
-  match (i.[0], parens_kind i) with
-  | '.', Some (s, o, c) -> Some (s, o, c)
+let split_index_op s =
+  match String.rindex s '.' with
+  | Some i when String.length s > i + 2 ->
+      let before = String.sub s ~pos:0 ~len:i in
+      let after = String.sub s ~pos:i ~len:(String.length s - i) in
+      Some ((if String.is_empty before then "" else "." ^ before), after)
   | _ -> None
 
+let index_op_get i =
+  match split_index_op i with
+  | None -> None
+  | Some (before, after) -> (
+    match parens_kind after with
+    | Some (s, o, c) -> Some (before ^ s, o, c)
+    | None -> None )
+
 let index_op_set i =
   match String.chop_suffix i ~suffix:"<-" with
   | None -> None
-  | Some i -> (
-    match (i.[0], parens_kind i) with
-    | '.', Some (s, o, c) -> Some (s, o, c)
-    | _ -> None )
+  | Some i -> index_op_get i
+
+let index_op_lid i ~f = f (String.concat ~sep:"" (Longident.flatten i))
```

Comment:
Why convert it to a string and then parse it in `split_index_op` ?
Why not inspect the long indent value (or the result of `Longident.flatten`) ?

Reply:
I reduced the diff, but it is indeed easier to convert the identifiers to strings and re-use the existing `index_op_get` function.